pettingzoo/classic/chess/chess_utils.py

```diff
@@ -324,18 +323,31 @@ def get_observation(orig_board, player: int):
       }
     """
     # from 0-63
-    # Adjust the row number for the white pawn to the 8th if the en passant flag is set, and vice versa for black pawns.
-    square = board.ep_square  # (int) where the en passant happened
+    # Adjust the row number for the white pawn to the 1st if the en passant flag is set, and vice versa for black pawns.
+    # For example
+    # If the white play an en passant move, the opponent can play a special move called en passant capture.
+    # To show this, we denote the pawn at (row, col) = (1, `dest_square`) instead of (5, `dest_square`).
```

Comment:
I don’t get why we denote it differently with 1, dest square. Is that just because it’s impossible for a pawn to go backwards so this is just the only encoding for a pawn being in en passant position? Is this something done by AlphaZero or others? Seems a bit odd.

Reply:
Yes, you are right, a white pawn never walks to the 1st row. It is just a notation of the chance to make a en passant capture. It is written in the document `pettingzoo/classic/chess/chess.py`#41. To my knowledge I have no idea whether it is conventional, but I just follow the definition already in pettinzoo.chess_v6.

Reply:
I review the paper of AlphaZero and the notation system of chess called Forsyth-Edwards Notation (FEN). AlphaZero mentioned some of special movement in chess like queen castling, but nothing about en passant capture. They did not explain but I guess the reason is a neural network can retrive information of en passant flag from their board history. On the other hand, FEN is a popular notation used on chess. I also see a function for it in the module `chess`. FEN use a individual string to denote whether a en passant flag is set or not, instead of putting into the 1st line of the board.

Back to our observation space, so far I revision the en passant section to keep its functionality. However, it is not conventional to my knowledge. In addition, we have provided a borad history where a model can retrive the information of en passant flag. Therefore, I would like to comment out this section of code.

Reply:
Could you check other RL chess implementations such as openspiel or other papers? Just want to make sure it won’t mess things up for people. I think I linked it previously but RLlib has a LeelaChessZero model, maybe we can check if that would work with this change? I can try to get in contact with the person who did the pr for that if need be

Reply:
OK, I will check and inform you if I need help.

Reply:
I copied this behavior from LC0, but I took a 2nd look at the AlphaZero paper and it definitely doesn't mention this. Presumably alphazero is relying on the board history to provide the necessary information for en-passant, and the Lc0 developers decided to add this information in explicitly.

Reply:
If the current code behavior is also done by leela chess zero that seems reasonable to do imo. Having some way of accessing the information is better than none. Do you have thoughts on the FEN system mentioned above @benblack769?

Reply:
Yeah, FEN's representation is nice because it is concise and unambiguous, but neural networks aren't necessarily the best at learning concise information, it can be better to spread them out.

For a code reference, here is how Lc0 decodes FEN into its internal representation of en-passant https://github.com/LeelaChessZero/lc0/blob/master/src/chess/board.cc#L1114